Add some docs about useful step before filing a PR

krnowak opened the pull request:
Also, CI asks about running `make precommit` if tree isn't clean.

It wasn't clear how to, for instance, regenerate a file after changing
one of the EventType constants.

akvanhar commented on README.md:
```diff
@@ -6,6 +6,14 @@
 
 The Go [OpenTelemetry](https://opentelemetry.io/) client.
 
+## Contributing
```
I notice most other repos have a CONTRIBUTING.md file.
Consider creating such a file and moving this text there.

rghetia replied:
@krnowak will you modify this PR to fix this or another PR?

krnowak replied:
I have changed this PR.

krnowak commented:
Updated. I have stolen the file from opentelemetry-python. Not sure if all the sections make sense here, though. Probably the "Become a Collaborator" section could basically be a link to opentelemetry community repo.

bogdandrutu commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,135 @@
+To create a new PR, fork the project in GitHub and clone the upstream repo:
+
+```sh
+$ git clone https://https://github.com/open-telemetry/opentelemetry-go.git
````
Do you not want to use go get for this? Or is that the old way?

rghetia replied:
go get is better as it clones it into $GOPATH/src directory.

krnowak replied:
I wouldn't say it's better - it's a matter of personal preference, especially that opentelemetry-go is using go modules, so it builds fine outside `${GOPATH}`. So, I'll provide both ways. Also, there is the `https://https://` to fix. ;)

krnowak commented:
Updated. Now it also recommends using either `go get` or `git clone`, fixed the `git clone` invocation (used to have `https://https://`), fixed adding a git remote (was using https link, instead of ssh), and replaced the "Becoming the collaborator/maintainer" part with a link to the document in the community repo.

rghetia commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,133 @@
+repo:
+
+```sh
+$ go get github.com/open-telemetry/opentelemetry-go
````
it should be go get go.opentelemetry.io other wise build fails as expected import path is go.opentelemetry.io.

krnowak replied:
Right, fixed.

rghetia commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,133 @@
+```sh
+$ git checkout -b <YOUR_BRANCH_NAME>
+# edit files
+$ make precommit
````
precommit will fail without necessary tools (golangci-lint, misspell, etc)
May be add install-tools target in Makefile.

akvanhar replied:
OOooh. An install-tools target would be great. I definitely ran into this when trying to get started the first time.

krnowak replied:
That rather sounds like a bug - I'd like to see an output from `make precommit` with a failure about the missing tools, because it should normally build the tools if they are missing. The `precommit` target depends on the every tool…

```
$ git clean -ffdx
$ make precommit
go build -o ./.tools/golangci-lint github.com/golangci/golangci-lint/cmd/golangci-lint
go build -o ./.tools/misspell github.com/client9/misspell/cmd/misspell
go build -o ./.tools/stringer golang.org/x/tools/cmd/stringer
PATH="/home/kv/projects/lightstep/opentelemetry-go/ci-unclean-msg/.tools:${PATH}" go generate ./...
```

So I'd prefer avoiding adding this.

rghetia replied:
```
➜  export GOPATH=/Users/rahulpa/rghetia/go/new
➜  go get go.opentelemetry.io
package go.opentelemetry.io: build constraints exclude all Go files in /Users/rahulpa/rghetia/go/new/src/go.opentelemetry.io
➜  cd $GOPATH/src/go.opentelemetry.io
➜  go.opentelemetry.io git:(master) make precommit
go build -o ./.tools/golangci-lint github.com/golangci/golangci-lint/cmd/golangci-lint
can't load package: package github.com/golangci/golangci-lint/cmd/golangci-lint: cannot find package "github.com/golangci/golangci-lint/cmd/golangci-lint" in any of:
	/usr/local/Cellar/go/1.12.4/libexec/src/github.com/golangci/golangci-lint/cmd/golangci-lint (from $GOROOT)
	/Users/rahulpa/rghetia/go/new/src/github.com/golangci/golangci-lint/cmd/golangci-lint (from $GOPATH)
make: *** [.tools/golangci-lint] Error 1
➜  go.opentelemetry.io git:(master)
```

krnowak replied:
Thanks. I can reproduce it with go 1.12, but not with go 1.13. Let's see if I can get some workaround.

krnowak replied:
Updated the PR again. Basically go older than 1.13 + repo in GOPATH = ignore modules. I added the commit to force the modules to be used. This should make things work.

krnowak commented:
Updated.

akvanhar commented:
This is great. I think it's really going to help future contributors.
